fe/fe-core/src/main/java/com/starrocks/sql/InsertPlanner.java

```diff
@@ -310,4 +316,41 @@ OptExprBuilder castOutputColumnsTypeToTargetColumns(ColumnRefFactory columnRefFa
         }
         return root.withNewRoot(new LogicalProjectOperator(new HashMap<>(columnRefMap)));
     }
+
+    /**
+     * OlapTableSink may be executed in multiply fragment instances of different machines
+     * For non-duplicate key types, we must guarantee that the orders of the same key are
+     * exactly the same. In order to achieve this goal, we can perform shuffle before TableSink
+     * so that the same key will be sent to the same fragment instance
+     */
+    private PhysicalPropertySet createPhysicalPropertySet(InsertStmt insertStmt,
+                                                          List<ColumnRefOperator> outputColumns) {
+        if (!(insertStmt.getTargetTable() instanceof OlapTable)) {
+            return new PhysicalPropertySet();
+        }
+
+        OlapTable table = (OlapTable) insertStmt.getTargetTable();
+
+        if (KeysType.DUP_KEYS.equals(table.getKeysType())) {
+            return new PhysicalPropertySet();
+        }
+
+        List<Column> columns = table.getFullSchema();
+        Preconditions.checkState(columns.size() == outputColumns.size(),
+                "outputColumn's size must equal with table's column size");
```

Comment:
If origin table columns size is 3, but one column has default value.
we can insert only two columns, like:
```
MySQL td> desc t1;
+-------+--------+------+------+---------+-------+
| Field | Type   | Null | Key  | Default | Extra |
+-------+--------+------+------+---------+-------+
| v1    | BIGINT | Yes  | true | <null>  |       |
| v2    | BIGINT | Yes  | true | <null>  |       |
| v3    | BIGINT | Yes  | true | <null>  |       |
+-------+--------+------+------+---------+-------+
3 rows in set
Time: 0.006s
MySQL td>
MySQL td> insert into t1(v1, v2) select v1, v2 from t2;
Query OK, 2 rows affected
Time: 0.131s
MySQL td> select * from t1;
+----+----+--------+
| v1 | v2 | v3     |
+----+----+--------+
| 1  | 10 | <null> |
| 2  | 20 | <null> |
+----+----+--------+
2 rows in set
Time: 0.018s
```

Reply:
The following statement will collect all the specified columns and un-specified nullable columns. So the condition is always true

```java
        //3. Fill in the default value and NULL
        OptExprBuilder optExprBuilder = fillDefaultValue(logicalPlan, columnRefFactory, insertStmt, outputColumns);
```